### PR TITLE
qbit nohardlinks fixes

### LINKED ISSUE #255: [Bug]: Type Error when `max_ratio` not set for a category under noHL
### Describe the Bug

When `max_ratio` is not defined for my `other` category, I get a type error when running qbit_manage:

```python
Traceback (most recent call last):
  File "/home/aaearon/qbit_manage/qbit_manage.py", line 544, in <module>
    start_loop()
  File "/home/aaearon/qbit_manage/qbit_manage.py", line 315, in start_loop
    start()
  File "/home/aaearon/qbit_manage/qbit_manage.py", line 411, in start
    num_tagged, num_untagged, num_deleted, num_deleted_contents = cfg.qbt.tag_nohardlinks()
  File "/home20/aaearon/qbit_manage/modules/qbittorrent.py", line 562, in tag_nohardlinks
    if tracker["max_ratio"] and tracker["max_ratio"] >= nohardlinks[category]["max_ratio"]:
TypeError: '>=' not supported between instances of 'float' and 'NoneType'
```

### Config

```yaml
# This is an example configuration file that documents all the options.
# It will need to be modified for your specific use case.
# Please refer to the link below for more details on how to set up the configuration file
# https://github.com/StuffAnThings/qbit_manage/wiki/Config-Setup

# commands:
#   # The commands defined below will IGNORE any commands used in command line and docker env variables.
#   dry_run: false
#   cross_seed: false
#   recheck: true
#   cat_update: false
#   tag_update: true
#   rem_unregistered: true
#   tag_tracker_error: true
#   rem_orphaned: true
#   tag_nohardlinks: true
#   skip_cleanup: false

qbt:
  # qBittorrent parameters
  host: HOSTNAME
  user: DELETE
  pass: DELETE

settings:
  force_auto_tmm: false  # Will force qBittorrent to enable Automatic Torrent Management for each torrent.
  tracker_error_tag: issue  # Will set the tag of any torrents that do not have a working tracker.
  ignoreTags_OnUpdate: # When running tag-update function, it will update torrent tags for a given torrent even if the torrent has one or more of the tags defined here.
  - noHL
  - issue
  - cross-seed
  nohardlinks_tag: noHL
directory:
  # Do not remove these
  # Cross-seed var: </your/path/here/>  # Output directory of cross-seed
  # root_dir var: </your/path/here/>  # Root downloads directory used to check for orphaned files, noHL, and RecycleBin.
  # <OPTIONAL> remote_dir var: </your/path/here/>  # Path of docker host mapping of root_dir.
  # Must be set if you're running qbit_manage locally and qBittorrent/cross_seed is in a docker
  # <OPTIONAL> recycle_bin var: </your/path/here/>   # Path of the RecycleBin folder. Default location is set to remote_dir/.RecycleBin
  # <OPTIONAL> torrents_dir var: </your/path/here/>  # Path of the your qbittorrent torrents directory. Required for `save_torrents` attribute in recyclebin
  cross_seed: /home/aaearon/downloads/qbittorrent/
  root_dir: /home/aaearon/downloads/qbittorrent/
  remote_dir: /home/aaearon/downloads/qbittorrent/
  orphaned_dir: /home/aaearon/downloads/qbittorrent/orphaned_data

  torrents_dir:
cat:
  # Category & Path Parameters
  # <Category Name> : <save_path>  # Path of your save directory.
  radarr: /home/aaearon/downloads/qbittorrent/movies
  radarr-complete: /home/aaearon/downloads/qbittorrent/movies
  tv-sonarr: /home/aaearon/downloads/qbittorrent/tv
  tv-sonarr-complete: /home/aaearon/downloads/qbittorrent/tv
  other: /home/aaearon/downloads/qbittorrent
  other.cross-seed: /home/aaearon/downloads/qbittorrent/other
  radarr.cross-seed: /home/aaearon/downloads/qbittorrent/movies
  tv-sonarr.cross-seed: /home/aaearon/downloads/qbittorrent/tv

cat_change:
  # This moves all the torrents from one category to another category. This executes on --cat-update
  # WARNING: if the paths are different and Default Torrent Management Mode is set to automatic the files could be moved !!!
  # <Old Category Name> : <New Category>
  radarr.cross-seed: other
  radarr-complete: other
  tv-sonarr-complete: other
  tv-sonarr.cross-seed: other
  other.cross-seed: other

tracker:
  # Tag Parameters
  # <Tracker URL Keyword>:    # <MANDATORY> This is the keyword in the tracker url
  # <MANDATORY> Set tag name. Can be a list of tags or a single tag
  #   tag: <Tag Name>
  # <OPTIONAL> Will set the torrent Maximum share ratio until torrent is stopped from seeding/uploading. -2 means the global limit should be used, -1 means no limit.
  #   max_ratio: 5.0
  # <OPTIONAL> Will set the torrent Maximum seeding time (min) until torrent is stopped from seeding. -2 means the global limit should be used, -1 means no limit.
  #   max_seeding_time: 129600
  # <OPTIONAL> Will ensure that noHL torrents from this tracker are not deleted by cleanup variable if torrent has not yet met the minimum seeding time (min).
  #   min_seeding_time: 2000
  # <OPTIONAL> Will limit the upload speed KiB/s (KiloBytes/second) (-1 means no limit)
  #   limit_upload_speed: 150
  # <OPTIONAL> Set this to the notifiarr react name. This is used to add indexer reactions to the notifications sent by Notifiarr
  #   notifiarr: <notifiarr indexer>
  # iptorrents
  empirehost:
    tag: iptorrents
    max_ratio: 1.05
    # 14 days
    max_seeding_time: 20170
    # min_seeding_time: 10
  # BroadcastTheNet
  landof.tv:
    tag: broadcasthenet
    # max_ratio: 1.05
    # 5 days
    # max_seeding_time: 7560
  # TorrentDay
  jumbohostpro:
    tag: torrentday
    max_ratio: 1.05
    # 3 days
    max_seeding_time: 4330
  # TorrentSeeds
  torrentseeds:
    tag: torrentseeds
    max_ratio: 1.05
    # 5 days
    max_seeding_time: 7300
  # Hawke-Uno
  hawke.uno:
    tag: huno
    # 5 days
    max_seeding_time: 7300
  # TorrentLeech
  torrentleech:
    tag: torrentleech
    max_ratio: 1.05
    # 8 days
    max_seeding_time: 11620
    min_seeding_time: 0
  # AlphaRatio
  alpharatio:
    tag: alpharatio
    max_ratio: 1.05
    # 7 days
    max_seeding_time: 10180
  # HDBits
  hdbits:
    tag: hdbits
  sportscult:
    tag: sportscult
  other:
    tag: other

nohardlinks:
  # Tag Movies/Series that are not hard linked
  # Mandatory to fill out directory parameter above to use this function (root_dir/remote_dir)
  # This variable should be set to your category name of your completed movies/completed series in qbit. Acceptable variable can be any category you would like to tag if there are no hardlinks found
  radarr:
    # <OPTIONAL> exclude_tags var: Will exclude the following tags when searching through the category.
    exclude_tags:
    # - hdbits
    # <OPTIONAL> cleanup var: WARNING!! Setting this as true Will remove and delete contents of any torrents that have a noHL tag and meets share limits
    cleanup: true
    # <OPTIONAL> max_ratio var: Will set the torrent Maximum share ratio until torrent is stopped from seeding/uploading
    # max_ratio: 4.0
    # <OPTIONAL> max seeding time var: Will set the torrent Maximum seeding time (min) until torrent is stopped from seeding
    # max_seeding_time: 86400
    # <OPTIONAL> Limit Upload Speed var: Will limit the upload speed KiB/s (KiloBytes/second) (`-1` : No Limit)
    # limit_upload_speed:
    # <OPTIONAL> min seeding time var: Will prevent torrent deletion by cleanup variable if torrent has not yet minimum seeding time (min).
    # Delete this key from a category's config to use the tracker's configured min_seeding_time. Will default to 0 if not specified for the category or tracker.
    # min_seeding_time: 43200
    # <OPTIONAL> resume_torrent_after_untagging_noHL var: If a torrent was previously tagged as NoHL and now has hardlinks, this variable will resume your torrent after changing share limits
    resume_torrent_after_untagging_noHL: true
  # Can have additional categories set with separate ratio/seeding times defined.
  tv-sonarr:
    # <OPTIONAL> exclude_tags var: Will exclude the following tags when searching through the category.
    exclude_tags:
    # - hdbits
    # <OPTIONAL> cleanup var: WARNING!! Setting this as true Will remove and delete contents of any torrents that have a noHL tag and meets share limits
    cleanup: true
    # <OPTIONAL> max_ratio var: Will set the torrent Maximum share ratio until torrent is stopped from seeding/uploading
    # max_ratio: 4.0
    # <OPTIONAL> max seeding time var: Will set the torrent Maximum seeding time (min) until torrent is stopped from seeding
    # max_seeding_time: 86400
    # <OPTIONAL> Limit Upload Speed var: Will limit the upload speed KiB/s (KiloBytes/second) (`-1` : No Limit)
    # limit_upload_speed:
    # <OPTIONAL> min seeding time var: Will prevent torrent deletion by cleanup variable if torrent has not yet minimum seeding time (min).
    # min_seeding_time: # Not specified for this category; tracker's value will be used. Will default to 0 if not specified for the category or tracker.
    # <OPTIONAL> resume_torrent_after_untagging_noHL var: If a torrent was previously tagged as NoHL and now has hardlinks, this variable will resume your torrent after changing share limits
    resume_torrent_after_untagging_noHL: true
  other:
    # max_ratio: 1.05
    cleanup: true
    exclude_tags:
      - hdbits
      - broadcasthenet
      - sportscult
    resume_torrent_after_untagging_noHL: true
    # min_seeding_time: 0

recyclebin:
  # Recycle Bin method of deletion will move files into the recycle bin (Located in /root_dir/.RecycleBin) instead of directly deleting them in qbit
  # By default the Recycle Bin will be emptied on every run of the qbit_manage script if empty_after_x_days is defined.
  enabled: false
  # <OPTIONAL> empty_after_x_days var:
  # Will automatically remove all files and folders in recycle bin after x days. (Checks every script run)
  # If this variable is not defined it, the RecycleBin will never be emptied.
  # WARNING: Setting this variable to 0 will delete all files immediately upon script run!
  empty_after_x_days: 0
  # <OPTIONAL> save_torrents var:
  # If this option is set to true you MUST fill out the torrents_dir in the directory attribute.
  # This will save a copy of your .torrent and .fastresume file in the recycle bin before deleting it from qbittorrent
  save_torrents: true
  # <OPTIONAL> split_by_category var:
  # This will split the recycle bin folder by the save path defined in the `cat` attribute
  # and add the base folder name of the recycle bin that was defined in the `recycle_bin` sub-attribute under directory.
  split_by_category: false

orphaned:
  # Orphaned files are those in the root_dir download directory that are not referenced by any active torrents.
  # Will automatically remove all files and folders in orphaned data after x days. (Checks every script run)
  # If this variable is not defined it, the orphaned data will never be emptied.
  # WARNING: Setting this variable to 0 will delete all files immediately upon script run!
  empty_after_x_days: 0
  # File patterns that will not be considered orphaned files. Handy for generated files that aren't part of the torrent but belong with the torrent's files
  exclude_patterns:
  - '**/.DS_Store'
  - '**/Thumbs.db'
  - '**/@eaDir'
  - /data/torrents/temp/**
  - '**/*.!qB'
  - '**/_unpackerred'

webhooks:
  error:
  run_start:
  run_end:
  function:
    cross_seed:
    recheck:
    cat_update:
    tag_update:
    rem_unregistered:
    tag_tracker_error:
    rem_orphaned:
    tag_nohardlinks:
    cleanup_dirs:
```

### Logs

https://gist.github.com/aaearon/fb927e319b1d1210a912ae36f4062ef8

### Screenshots

_No response_

### Installation

Local

### Version Number

3.5.0

### What branch are you on?

master

### PR DESCRIPTION
**Pull request by bakerboy448**
## Description
- (fix) qbit nohardlinks ratio/seed time if tracker not set
- (fix) qbit nohardlinks ratio/seed time if tracker not set

Fixes #255

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the docstring for new or existing methods
- [x] I have added tests when applicable

### CODE REVIEW

File: modules/qbittorrent.py [hunk 1]
```diff
@@ -547,36 +547,41 @@ def add_tag_no_hl(add_tag=True):
...
                             if (
-                                tracker["min_seeding_time"]
+                                tracker["min_seeding_time"] is not None
                                 and tracker["min_seeding_time"] >= nohardlinks[category]["min_seeding_time"]
                             ):
                                 min_seeding_time = tracker["min_seeding_time"]
                             elif nohardlinks[category]["min_seeding_time"]:
                                 min_seeding_time = nohardlinks[category]["min_seeding_time"]
+                           else:
+                            	min_seeding_time = min_seeding_time
```
Comment (bobokun) [inline modules/qbittorrent.py:557]: This else statement is not needed as we assign `min_seeding_time` in line 548

File: modules/qbittorrent.py [hunk 2]
```diff
@@ -547,36 +547,41 @@ def add_tag_no_hl(add_tag=True):
...
                             # If both tracker and nohardlinks category setting is set, use the larger of the two
                             # If neither set, use -1 (no limit)
                             max_ratio = -1
-                            if tracker["max_ratio"] and tracker["max_ratio"] >= nohardlinks[category]["max_ratio"]:
+                            if tracker["max_ratio"] is not None and tracker["max_ratio"] >= nohardlinks[category]["max_ratio"]:
                                 max_ratio = tracker["max_ratio"]
                             elif nohardlinks[category]["max_ratio"]:
                                 max_ratio = nohardlinks[category]["max_ratio"]
+                            else:
+                            	max_ratio = max_ratio
```
Comment (bobokun) [inline modules/qbittorrent.py:569]: same here

File: modules/qbittorrent.py [hunk 3]
```diff
@@ -547,36 +547,41 @@ def add_tag_no_hl(add_tag=True):
...
-                                tracker["max_seeding_time"]
+                                tracker["max_seeding_time"] is not None
                                 and tracker["max_seeding_time"] >= nohardlinks[category]["max_seeding_time"]
                             ):
                                 max_seeding_time = tracker["max_seeding_time"]
                             elif nohardlinks[category]["max_seeding_time"]:
                                 max_seeding_time = nohardlinks[category]["max_seeding_time"]
-
+                            else:
+                            	max_seeding_time = max_seeding_time
```
Comment (bobokun) [inline modules/qbittorrent.py:584]: same here

**Review by bobokun — CHANGES_REQUESTED** [review]
some minor changes

**Comment by bobokun** [comment]
Hmm, can we make a PR to dev first?

**Comment by bakerboy448** [comment]
Switched to dev

**Review by bobokun — APPROVED** [review]

**Pull request merged**